HW1/P2/P2.txt

```diff
@@ -1 +1,3 @@
 # Your discussion here
+
+Fromt the two histogram plots we can see that the load of workers in P2b is much more balanced than that of P2a, as the total iteration time for each partition is narrowed in a small range. However, the transform of repartition will result in shuffle, which causes much more overhead in computation time. So this is the tradeoff.
```

Comment:
The most elegant solution for load balancing I saw was someone who took the computational grid and assigned individual pixels randomly to partitions, so that each partition wound up with a roughly equal number of computationally heavy and light points. This randomization had to be done before converting the grid to RDD.